### DISCUSSION TITLE
Saving a checkpoint at each iteration

### DISCUSSION THREAD

**Discussion by Maltergate** [PasaOpasen/geneticalgorithm2-6.8.7#60] [Ideas]
**EDIT:** My bad, just saw the callbacks. I saw only the middle callbacks before.

Currently, it's possible to save the last_generation (in the run command) to save the population and potentially restart a new optimization. However, it'd be nice to checkpoint the population/scores etc at each iteration so that, if the algorithm is killed (like running out of time on HPC for example, or need the computing resources for something else), then we can restart the optimization "where we left off" by specifying the last checkpoint.

That would be a great feature and would definitively make the package more HPC ready.

Thanks

**Comment by PasaOpasen** [comment]
@Maltergate 

There are middle_callbacks https://github.com/PasaOpasen/geneticalgorithm2#middle-callbacks. With them u can do everything u want, save collect data per population, save each generation and so on

Of course, u should write your own function for this, but it's easy

if the problem is not so complicated, u can use more simple callbacks
